src/gtk/srchctrl.cpp

```diff
@@ -0,0 +1,429 @@
+    return true;
+}
+
+void wxSearchCtrl::GTKCreateSearchEntryWidget()
+{
+#if wxHAS_GTK_SEARCH_ENTRY
+    if ( wx_is_at_least_gtk3(6) )
+    {
+        m_widget = gtk_search_entry_new();
+    }
+#endif // wxHAS_GTK_SEARCH_ENTRY
+
+    if ( !m_widget )
+    {
+        // No GtkSearchEntry! fallback to the plain GtkEntry.
+        m_widget = gtk_entry_new();
+    }
+
+    g_object_ref(m_widget);
+
+    m_entry = GTK_ENTRY(m_widget);
+    g_object_add_weak_pointer(G_OBJECT(m_entry), (void**)&m_entry);
+
+    if ( !wx_is_at_least_gtk3(6) )
+    {
+        gtk_entry_set_icon_from_icon_name(m_entry, 
+                                          GTK_ENTRY_ICON_PRIMARY,
+                                          "edit-find-symbolic");
+
+        // Mimic the behaviour of the native GtkSearchEntry.
+        gtk_entry_set_icon_sensitive(m_entry, GTK_ENTRY_ICON_PRIMARY, FALSE);
+        gtk_entry_set_icon_activatable(m_entry, GTK_ENTRY_ICON_PRIMARY, FALSE);
+    }
+
+    g_signal_connect(m_entry, "icon-press", G_CALLBACK(wx_gtk_icon_press), this);
+}
+
+GtkEditable *wxSearchCtrl::GetEditable() const
+{
+    return GTK_EDITABLE(m_entry);
+}
+
+void wxSearchCtrl::Clear()
+{
+    wxTextEntry::Clear();
+    ShowCancelButton(false);
+}
+
+bool wxSearchCtrl::IsModified() const
+{
+    return m_modified;
+}
+
+void wxSearchCtrl::MarkDirty()
+{
+    m_modified = true;
+}
+
+void wxSearchCtrl::DiscardEdits()
+{
+    m_modified = false;
+}
+
+bool wxSearchCtrl::PositionToXY(long pos, long *x, long *y ) const
+{
+    if (pos <= GTKGetEntryTextLength(GetEntry()))
+    {
+        if ( y )
+            *y = 0;
+        if ( x )
+            *x = pos;
+    }
+    else
+    {
+        // index out of bounds
+        return false;
+    }
+
+    return true;
+}
+
+long wxSearchCtrl::XYToPosition(long x, long y ) const
+{
+    if ( y != 0 || x > GTKGetEntryTextLength(GetEntry()) )
+        return -1;
+
+    return x;
+}
+
+int wxSearchCtrl::GetLineLength(long lineNo) const
+{
+    const wxString str = GetLineText(lineNo);
+    return (int) str.length();
+}
+
+int wxSearchCtrl::GetNumberOfLines() const 
+{
+    return 1;
+}
+
+wxString wxSearchCtrl::GetLineText( long lineNo ) const
+{
+    if ( lineNo == 0 )
+        return GetValue();
+
+    return wxString();
+}
+
+void wxSearchCtrl::ShowPosition( long pos )
+{
+    gtk_editable_set_position(GetEditable(), pos);
+}
+
+// search control specific interfaces
+// ----------------------------------
+#if wxUSE_MENUS
+
+void wxSearchCtrl::SetMenu( wxMenu* menu )
+{
+    if ( menu == m_menu )
+    {
+        // no change
+        return;
+    }
+
+    delete m_menu;
+    m_menu = menu;
+
+    const bool hasMenu = m_menu != NULL;
+
+    gtk_entry_set_icon_sensitive(m_entry, GTK_ENTRY_ICON_PRIMARY, hasMenu);
+    gtk_entry_set_icon_activatable(m_entry, GTK_ENTRY_ICON_PRIMARY, hasMenu);
+}
+
+wxMenu* wxSearchCtrl::GetMenu()
+{
+    return m_menu;
+}
+
+#endif // wxUSE_MENUS
+
+void wxSearchCtrl::ShowSearchButton(bool WXUNUSED(show))
+{
```

Comment:
It would be nice to add a comment here explaining why we don't implement this, e.g. just a simple `// Search button is always shown in the native control` if this is the reason.

Reply:
Yes this is the intention.
and for completeness i made 
`wxSearchCtrl::IsSearchButtonVisible()` returns true unconditionally.